etcd/server.sls

```diff
@@ -4,12 +4,8 @@
 etcd_packages:
   pkg.installed:
   - names: {{ server.pkgs }}
-{%- if server.get('engine', 'systemd') %}
-  - require:
-    - file: /etc/default/etcd
-{%- endif %}
 
-{%- if server.get('engine', 'systemd') == 'kubernetes' %}
+{%- if server.engine == 'kubernetes' %}
```

Comment:
Defaults should be defined in map.jinja or using `get('xyz', default)` or `|default(default)` jinja filter.